src/rdf4cpp/rdf/datatypes/owl/Rational.cpp

```diff
@@ -0,0 +1,54 @@
+#include <rdf4cpp/rdf/datatypes/owl/Rational.hpp>
+
+namespace rdf4cpp::rdf::datatypes::registry {
+
+template<>
+capabilities::Default<owl_rational>::cpp_type capabilities::Default<owl_rational>::from_string(std::string_view s) {
+    if (auto pos = s.find_last_of('-'); pos != std::string_view::npos && pos != 0) {
+        // owl:rational only allows - at beginning
+        throw std::runtime_error{"owl:rational parsing error: invalid sign position"};
+    }
+
+    if (s.find_first_not_of("0123456789/-") != std::string_view::npos) {
+        throw std::runtime_error{"owl:rational parsing error: invalid character in string"};
+    }
+
+    try {
+        return cpp_type{s};
+    } catch (std::runtime_error const &e) {
+        throw std::runtime_error{std::string{"owl:rational parsing error:"} + e.what()};
+    }
+}
+
+template<>
+std::string capabilities::Default<owl_rational>::to_string(cpp_type const &value) {
+    if (auto den = denominator(value); den < 0) {
```

Comment:
add small example that explains what happens here

Reply:
basically, the reason is that `boost::multiprecision::cpp_rational` allows values such as `1/-5`. Which is invalid for `owl:rational`.